woocommerce/woocommerce
# Fix/Additional checkout field  select type placeholder not shown through edit account page form

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:




This PR prepares the options for a select box used as an additional address field, so it can display a placeholder. On that view, opposed to the checkout block, it doesn't rely on the data placeholder entry, and instead expects an empty keyed placeholder text.

Closes #52110 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Add the following snippet to your theme functions.php or at the end of the main woocommerce.php file.
2. Go to My Account > Addresses > Edit Shipping Address
3. Verify that the placeholder "Click to choose something" is selected. Save other values and verify they get persisted.
5. Change back to the placeholder, and verify the value is cleared.

```php
add_action(
	'woocommerce_init',
	function() {
		woocommerce_register_additional_checkout_field(
			array(
				'id'            => 'mynamespace/number',
				'type'          => 'select',
				'label'         => 'Choose something',
				'location'      => 'address',
				'required'      => false,
				'options'       => [
					[
						'value' => 0,
						'label' => "ZERO"
					],
					[
						'value' => "1",
						'label' => "One!"
					],
					[
						'value' => 2,
						'label' => "Two!"
					],
					[
						'value' => 'green',
						'label' => "The color green!"
					],
					[
						'value' => "7",
						'label' => "Seven!"
					]
				],
				'placeholder'   => 'Click to choose something'
			),
		);
	}
);
```



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Fixed a bug causing placeholder for additional select fields not to be shown when editing addresses on My Account screen.
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
